public/js/reducers/sources.js

```diff
@@ -124,8 +132,15 @@ function _updateText(state, action : any) : Record<SourcesState> {
   }));
 }
 
-function removeSourceFromTabList(state, id) {
-  return state.tabs.filter(tab => tab.get("id") != id);
+function removeSourceFromTabList(tabs, id) {
+  return tabs.filter(tab => tab.get("id") != id);
+}
+
+function removeSourcesFromTabList(tabs, ids) {
+  ids.forEach(id => {
+    tabs = removeSourceFromTabList(tabs, id);
+  });
```

Comment:
hmm, this might be better expressed in terms of reduce

```js
return ids.reduce((id, tabs) => removeSourceFromTabList(tabs, id), tabs)
```

I'm not 100% sure, but an idea as the `tabs` assignment seems a bit strange here. Also, not a big deal.